From 06b56a476a77806f140b29b5e92d2db607302f09 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Sun, 6 Mar 2022 08:27:37 +0100
Subject: [PATCH 12/13] Setup eqos for RMII

Configure the eqos pinctrl for RMII, clock and set the eqos phy-mode to RMII.

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 arch/arm/dts/imx8mp-irma6r2.dts                 | 11 +++--------
 board/freescale/imx8mp_irma6r2/imx8mp_irma6r2.c | 14 ++++++++------
 2 files changed, 11 insertions(+), 14 deletions(-)

diff --git a/arch/arm/dts/imx8mp-irma6r2.dts b/arch/arm/dts/imx8mp-irma6r2.dts
index f1700743c6..5d37f006b4 100644
--- a/arch/arm/dts/imx8mp-irma6r2.dts
+++ b/arch/arm/dts/imx8mp-irma6r2.dts
@@ -90,7 +90,7 @@
 &eqos {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_eqos>;
-	phy-mode = "rgmii-id";
+	phy-mode = "rmii";
 	phy-handle = <&ethphy0>;
 	status = "okay";
 
@@ -385,17 +385,12 @@
 			MX8MP_IOMUXC_ENET_MDIO__ENET_QOS_MDIO	0x3
 			MX8MP_IOMUXC_ENET_RD0__ENET_QOS_RGMII_RD0	0x91
 			MX8MP_IOMUXC_ENET_RD1__ENET_QOS_RGMII_RD1	0x91
-			MX8MP_IOMUXC_ENET_RD2__ENET_QOS_RGMII_RD2	0x91
-			MX8MP_IOMUXC_ENET_RD3__ENET_QOS_RGMII_RD3	0x91
-			MX8MP_IOMUXC_ENET_RXC__CCM_ENET_QOS_CLOCK_GENERATE_RX_CLK	0x91
 			MX8MP_IOMUXC_ENET_RX_CTL__ENET_QOS_RGMII_RX_CTL	0x91
 			MX8MP_IOMUXC_ENET_TD0__ENET_QOS_RGMII_TD0		0x1f
 			MX8MP_IOMUXC_ENET_TD1__ENET_QOS_RGMII_TD1		0x1f
-			MX8MP_IOMUXC_ENET_TD2__ENET_QOS_RGMII_TD2		0x1f
-			MX8MP_IOMUXC_ENET_TD3__ENET_QOS_RGMII_TD3		0x1f
+			MX8MP_IOMUXC_ENET_TD2__CCM_ENET_QOS_CLOCK_GENERATE_REF_CLK 0x4000001f
 			MX8MP_IOMUXC_ENET_TX_CTL__ENET_QOS_RGMII_TX_CTL	0x1f
-			MX8MP_IOMUXC_ENET_TXC__CCM_ENET_QOS_CLOCK_GENERATE_TX_CLK	0x1f
-			MX8MP_IOMUXC_SAI2_RXC__GPIO4_IO22		0x19
+			MX8MP_IOMUXC_SAI1_TXD2__GPIO4_IO14		0x19
 		>;
 	};
 
diff --git a/board/freescale/imx8mp_irma6r2/imx8mp_irma6r2.c b/board/freescale/imx8mp_irma6r2/imx8mp_irma6r2.c
index cae451aed8..e03aba44fe 100644
--- a/board/freescale/imx8mp_irma6r2/imx8mp_irma6r2.c
+++ b/board/freescale/imx8mp_irma6r2/imx8mp_irma6r2.c
@@ -141,9 +141,9 @@ static int setup_fec(void)
 
 #ifdef CONFIG_DWC_ETH_QOS
 
-#define EQOS_RST_PAD IMX_GPIO_NR(4, 22)
+#define EQOS_RST_PAD IMX_GPIO_NR(4, 14)
 static iomux_v3_cfg_t const eqos_rst_pads[] = {
-	MX8MP_PAD_SAI2_RXC__GPIO4_IO22 | MUX_PAD_CTRL(NO_PAD_CTRL),
+	MX8MP_PAD_SAI1_TXD2__GPIO4_IO14 | MUX_PAD_CTRL(NO_PAD_CTRL),
 };
 
 static void setup_iomux_eqos(void)
@@ -165,12 +165,14 @@ static int setup_eqos(void)
 
 	setup_iomux_eqos();
 
-	/* set INTF as RGMII, enable RGMII TXC clock */
+	/* set INTF as RMII */
 	clrsetbits_le32(&gpr->gpr[1],
-			IOMUXC_GPR_GPR1_GPR_ENET_QOS_INTF_SEL_MASK, BIT(16));
-	setbits_le32(&gpr->gpr[1], BIT(19) | BIT(21));
+			IOMUXC_GPR_GPR1_GPR_ENET_QOS_INTF_SEL_MASK, BIT(18));
 
-	return set_clk_eqos(ENET_125MHZ);
+	/* GPR_ENET_QOS_CLK_GEN_EN (Bit19), IOMUXC_GPR_ENET_QOS_CLK_TX_CLK_SEL (Bit20)=1 loopback*/
+	setbits_le32(&gpr->gpr[1], BIT(19) | BIT(20));
+
+	return set_clk_eqos(ENET_50MHZ);
 }
 #endif
 
-- 
2.35.1

